core-bundle/contao/drivers/DC_Table.php

```diff
@@ -5418,7 +5418,7 @@ protected function searchMenu()
 				$option_label = \is_array($GLOBALS['TL_LANG']['MSC'][$field]) ? $GLOBALS['TL_LANG']['MSC'][$field][0] : $GLOBALS['TL_LANG']['MSC'][$field];
 			}
 
-			$options_sorter[$option_label . '_' . $field] = '  <option value="' . StringUtil::specialchars($field) . '"' . ((isset($session['search'][$this->strTable]['field']) && $session['search'][$this->strTable]['field'] == $field) ? ' selected="selected"' : '') . '>' . $option_label . '</option>';
+			$options_sorter[$option_label . '_' . $field] = '  <option value="' . StringUtil::specialchars($field) . '"' . (((isset($session['search'][$this->strTable]['field']) && $session['search'][$this->strTable]['field'] == $field) || (!isset($session['search'][$this->strTable]['field']) && ($GLOBALS['TL_DCA'][$this->strTable]['list']['sorting']['defaultSearchField'] ?? null) == $field)) ? ' selected="selected"' : '') . '>' . $option_label . '</option>';
```

Comment:
```suggestion
			$options_sorter[$option_label . '_' . $field] = '  <option value="' . StringUtil::specialchars($field) . '"' . ((($session['search'][$this->strTable]['field'] ?? $GLOBALS['TL_DCA'][$this->strTable]['list']['sorting']['defaultSearchField'] ?? null) == $field) ? ' selected="selected"' : '') . '>' . $option_label . '</option>';
```

Simplified?

Reply:
Did you test this? Because the original logic is

```php
if (isset($session['search'][$this->strTable]['field']) && $session['search'][$this->strTable]['field'] == $field) {
    // case A
} elseif (!isset($session['search'][$this->strTable]['field']) && ($GLOBALS['TL_DCA'][$this->strTable]['list']['sorting']['defaultSearchField'] ?? null) == $field) {
    // case B
}
```

and I‘m not sure your code does the same. It might though… 😄 

Here is what you should test:

1. Reset the session data in the back end profile menu.
2. Open the "Pages" module. → The search field should be `pageTitle`.
3. Change the search field to `description`, enter a value and submit. → The search field should be `description`.
4. Remove the value but leave the search field on `description` and submit. → The search field should still be `description`, because the session value overrides the default value.

Reply:
Yes, works the same way.
_(but I’m also fine if you want to keep the code as it is...)_